### ISSUE TITLE
local execution under a user running fish shell fails to gather facts

### ISSUE THREAD

**Issue by retrogradeorbit** [epiccastle/spire#52]
`spire -e '(local (get-fact))'`

```
Exception in thread "main" clojure.lang.ExceptionInfo: Assert failed: retrieving paths script exited 2: sh: 1: Syntax error: "(" unexpected(zero? exit) [at line 1, column 1] {:type :sci/error, :line 1, :column 1, :message "Assert failed: retrieving paths script exited 2: sh: 1: Syntax error: \"(\" unexpected\n\n(zero? exit) [at line 1, column 1]"}
	at sci.impl.utils$rethrow_with_location_of_node.invokeStatic(utils.cljc:74)
	at sci.impl.utils$rethrow_with_location_of_node.invoke(utils.cljc:58)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:518)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:498)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:547)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:530)
	at sci.impl.interpreter$eval_form.invokeStatic(interpreter.cljc:588)
	at sci.impl.interpreter$eval_form.invoke(interpreter.cljc:579)
	at sci.impl.interpreter$eval_string_STAR_.invokeStatic(interpreter.cljc:598)
	at sci.impl.interpreter$eval_string_STAR_.invoke(interpreter.cljc:593)
	at sci.impl.interpreter$eval_string$fn__4572.invoke(interpreter.cljc:610)
	at sci.impl.interpreter$eval_string.invokeStatic(interpreter.cljc:607)
	at sci.impl.interpreter$eval_string.invoke(interpreter.cljc:603)
	at sci.core$eval_string.invokeStatic(core.cljc:178)
	at sci.core$eval_string.invoke(core.cljc:141)
	at spire.eval$evaluate.invokeStatic(eval.clj:79)
	at spire.eval$evaluate.invoke(eval.clj:69)
	at spire.core$_main.invokeStatic(core.clj:73)
	at spire.core$_main.doInvoke(core.clj:56)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at spire.core.main(Unknown Source)
Caused by: java.lang.AssertionError: Assert failed: retrieving paths script exited 2: sh: 1: Syntax error: "(" unexpected(zero? exit)
	at spire.facts$run_and_return_lines.invokeStatic(facts.clj:180)
	at spire.facts$run_and_return_lines.invoke(facts.clj:178)
	at spire.facts$fn__1665.invokeStatic(facts.clj:231)
	at spire.facts$fn__1665.invoke(facts.clj:222)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at spire.facts$fetch_facts.invokeStatic(facts.clj:176)
	at spire.facts$fetch_facts.invoke(facts.clj:175)
	at spire.facts$update_facts_BANG_.invokeStatic(facts.clj:310)
	at spire.facts$update_facts_BANG_.invoke(facts.clj:309)
	at spire.facts$get_fact.invokeStatic(facts.clj:317)
	at spire.facts$get_fact.doInvoke(facts.clj:313)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at sci.impl.interpreter$fn_call.invokeStatic(interpreter.cljc:463)
	at sci.impl.interpreter$fn_call.invoke(interpreter.cljc:463)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:514)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:498)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:547)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:530)
	at sci.impl.interpreter$eval_do_STAR_$fn__4340.invoke(interpreter.cljc:420)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:420)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:417)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:430)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:467)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:465)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:505)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:498)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:547)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:530)
	at sci.impl.interpreter$eval_do_STAR_$fn__4340.invoke(interpreter.cljc:420)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:420)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:417)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:430)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:467)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:465)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:505)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:498)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:547)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:530)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:285)
	at sci.impl.interpreter$eval_try.invoke(interpreter.cljc:280)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:548)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:530)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:471)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:465)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:505)
	... 18 more  
```

looks like fish shell is detected but then script is run under /bin/sh by facts/runner

**Comment by PrestanceDesign** [comment]
Let me know, if I can give an missing information to help to reproduce/debug?

**Comment by retrogradeorbit** [comment]
fixed in 0.1.0-alpha.10

**Comment by PrestanceDesign** [comment]
Nice, thx for this new release!
